File: VerifyPreorderSerializationOfABinaryTree/VerifyPreorderSerializationOfABinaryTree.py
class Solution(object):
    def isValidSerialization(self, preorder):
        """
        :type preorder: str
        :rtype: bool
        """
        nullCnt = 0
        nodeCnt = 0
        preorderList = preorder.split(',')
        for i in xrange(len(preorderList)):
        	if preorderList[i] != '#':
        		nodeCnt += 1
        	else:
        		nullCnt += 1
        	if nullCnt >= nodeCnt + 1 and i != len(preorderList) - 1:
        		return False
        return nullCnt == nodeCnt + 1

File: VerifyPreorderSerializationOfABinaryTree/VerifyPreorderSerializationOfABinaryTree.cpp
class Solution {
public:
    bool isValidSerialization(string preorder) {
        istringstream iss(preorder);
        string tmp;
        vector<string> vec;
        while(getline(iss, tmp, ','))
        	vec.push_back(tmp);
        int nullCnt = 0, nodeCnt = 0;        
        for(int i = 0;i < vec.size(); i++) {
        	if(vec[i] == "#")
        		nullCnt++;
        	else
        		nodeCnt++;
        	if(nullCnt >= nodeCnt + 1 && i != vec.size() - 1)
        		return false;
        }
        return nullCnt == nodeCnt + 1;
    }
};